### PR TITLE
modification for flutter web auth 0.4.0

### PR DESCRIPTION
**Pull request by Silfalion**
These modifications allow taking advantage of flutter_web_auth new features for flutter web which is to open a new tab for authentication instead of using the in-use one.

This pull request includes:
- bumping flutter web auth version to 0.4.0.
- adding the cookijar variable as well as imports in client_browser.dart.
- implementing web_auth in client_browser.dart.
- removing the condition at the end of the createOAuth2Session method in account.dart.

### CODE REVIEW

File: templates/flutter/lib/src/client_browser.dart.twig [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import 'dart:html' as html;
+import 'dart:io';
```
Comment (lohanidamodar) [inline templates/flutter/lib/src/client_browser.dart.twig:2]: As this is browser_client, we should not use dart:io, cookie_jar, here, we should handle this some other way

File: templates/flutter/lib/src/client_browser.dart.twig [hunk 2]
```diff
@@ -128,6 +133,23 @@ class ClientBrowser extends ClientBase with ClientMixin {
...
+        throw AppwriteException(
+            "Invalid OAuth2 Response. Key and Secret not available.", 500);
+      }
+      Cookie cookie = new Cookie(key, secret);
+      cookie.domain = Uri.parse(_endPoint).host;
+      cookie.httpOnly = true;
+      cookie.path = '/';
+      List<Cookie> cookies = [cookie];
+      await init();
+      _cookieJar.saveFromResponse(Uri.parse(_endPoint), cookies);
```
Comment (lohanidamodar) [inline templates/flutter/lib/src/client_browser.dart.twig:153]: As this is web, we don't need to handle cookie ourselves, we can just remove this, with this we will not need to import cookie jar

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Hey thank you for the PR,
Let's handle these changes 🙏🏻

**Comment by lohanidamodar** [comment]
Also seems there is some conflict, please sync with master and fix it

File: templates/flutter/lib/src/client_browser.dart.twig (hunk 2 above)
Comment (Silfalion) [reply templates/flutter/lib/src/client_browser.dart.twig]: Deleted cookie jar and dart io altogether

**Comment by Silfalion** [comment]
Fixed the remaining conflicts for the pubspec.yaml file but the remaining one is about flutter_web_auth bump to 0.4.0

File: templates/flutter/lib/services/service.dart.twig [hunk 3]
```diff
@@ -59,12 +59,8 @@ class {{ service.name | caseUcfirst }} extends Service {
           query: query.join('&')
         );
 
-        if(kIsWeb) {
-          redirect(url.toString());
-          return Future.value();
-        }else{
-          return client.webAuth(url);
-        }
+
```
Comment (lohanidamodar) [inline templates/flutter/lib/services/service.dart.twig:62]:
```suggestion
```

File: templates/flutter/lib/src/client_browser.dart.twig [hunk 4]
```diff
@@ -19,6 +21,7 @@ class ClientBrowser extends ClientBase with ClientMixin {
   Map<String, String>? _headers;
   late Map<String, String> config;
   late BrowserClient _httpClient;
+
```
Comment (lohanidamodar) [inline templates/flutter/lib/src/client_browser.dart.twig:24]:
```suggestion
```

File: templates/flutter/pubspec.yaml.twig [hunk 5]
```diff
@@ -12,7 +12,7 @@ dependencies:
     sdk: flutter
   cookie_jar: ^3.0.1
   device_info_plus: ^3.1.1
-  flutter_web_auth: ^0.3.1
+  flutter_web_auth: ^0.4.0
   http: ^0.13.4
   package_info_plus: ^1.3.0
   path_provider: ^2.0.7
```
Comment (lohanidamodar) [inline templates/flutter/pubspec.yaml.twig:18]:
```suggestion
  path_provider: ^2.0.8
```

**Comment by lohanidamodar** [comment]
@Silfalion just removed some extra white spaces, and updated path_provider. Will let the tests pass and then merge.

**Review by lohanidamodar — APPROVED** [review]

**Comment by Silfalion** [comment]
@lohanidamodar awesome, thank you!

File: templates/flutter/lib/src/client_browser.dart.twig [hunk 6]
```diff
@@ -1,4 +1,6 @@
 import 'dart:html' as html;
+import 'package:cookie_jar/cookie_jar.dart';
```
Comment (lohanidamodar) [inline templates/flutter/lib/src/client_browser.dart.twig:2]:
```suggestion
```

**Review by wess — APPROVED** [review]

**Pull request merged**